requests_kerberos/kerberos_.py

```diff
@@ -71,6 +90,9 @@ def _negotiate_value(response):
         regex = re.compile('(?:.*,)*\s*Negotiate\s*([^,]*),?', re.I)
         _negotiate_value.regex = regex
 
+    if response is None: # we're doing an initial request- just use an empty negotiate value
+        return ''
+
```

Comment:
Let's re-evaluate this a little bit.

The purpose of this short-circuit execution, as far as I can tell, is to ensure that `generate_request_header` can build the pre-emptive auth as needed by calling this with a `None` response. However, you've accidentally (presumably as part of a refactor) left a boolean kwarg in `generate_request_header` that is unused.

I wonder if, rather than put the short-circuit all the way down here (in an arguably unrelated bit of code), the higher level function (`generate_request_header`) shouldn't just skip the the call to this function based on the knowledge it has. That would avoid some overhead (a function call and possibly some regex building), and also compartmentalises the logic a lot better.

It's actually orthogonal to the kwarg (you can still signal this with a `None` response), but it seems like that approach would be better. Do you disagree?